libs/http-ext/src/lib/matchers/match-origin/get-origin.ts

```diff
@@ -1,4 +1,21 @@
-export const getOrigin = (url: string): string => {
-  const [protocol,, host] = url.split(/\/|\?/);
-  return `${protocol}//${host}`;
+export function invalidUrlError(url: string) {
+  return new Error(`InvalidUrlError: ${url} is not a valid URL.`);
 }
+
+export const getOrigin = (url: string): string => {
+  const [scheme, urlWithoutScheme] = url.split('://');
+
+  if (scheme == null || urlWithoutScheme == null) {
+    throw invalidUrlError(url);
+  }
+
+  const result = urlWithoutScheme.match(/^[a-z0-9.:-]+/);
+
+  if (result == null) {
+    throw invalidUrlError(url);
+  }
+
+  const originWithoutScheme = result[0];
+
+  return `${scheme}://${originWithoutScheme}`;
+};
```

Comment:
I thought it might be safer to use a whitelist for the `fqdn:port` instead of a "blacklist" with `/` and `?` to find the limit.
How does it sound?

Reply:
Seems more reasonable! :+1: